daal4py/sklearn/_utils.py

```diff
@@ -411,3 +378,66 @@ def method_wrapper(self, *args, **kwargs):
         return result
 
     return method_wrapper
+
+
+def control_n_jobs(decorated_methods: list):
+    """Decorator for the control of 'n_jobs' parameter in estimator class. It applied
+    for all estimators with and without support of parameter in original sklearn.
+    In case of estimator without 'n_jobs' support, this decorator adds it.
+    Methods listed in 'decorated_methods' argument are only methods
+    to be run with 'n_jobs'.
+    """
```

Comment:
I think we can improve this description

```suggestion
"""Decorator for controlling the 'n_jobs' parameter in an estimator class.

This decorator is designed to be applied to both estimators with and without native support for the 'n_jobs' parameter in the original scikit-learn implementation. When applied to an estimator without 'n_jobs' support in its original '__init__' method, this decorator adds the 'n_jobs' parameter.

Additionally, this decorator allows for fine-grained control over which methods should be executed with the 'n_jobs' parameter. The methods specified in the 'decorated_methods' argument will run with 'n_jobs', while all other methods remain unaffected.

Parameters:
    decorated_methods (list): A list of method names to be executed with 'n_jobs'.

Example:
    @control_n_jobs(decorated_methods=['fit', 'predict'])
    class MyEstimator:
        def __init__(self, *args, **kwargs):
            # Your original __init__ implementation here

        def fit(self, *args, **kwargs):
            # Your original fit implementation here

        def predict(self, *args, **kwargs):
            # Your original predict implementation here

        def other_method(self, *args, **kwargs):
            # Methods not listed in decorated_methods will not be affected by 'n_jobs'
            pass
"""
```

And why not call the decorator simply `n_jobs_decorator`?

Reply:
Description was extended.

Reply:
It's not common to name decorators as `*_decorator` in sklearn at least.